ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -194,6 +194,8 @@ private[server] object ServerHelpers extends ServerHelpersPlatform {
         val handler: Stream[F, Nothing] = shutdown.trackConnection >>
           Stream
             .resource(upgradeSocket(connect, tlsInfoOpt, logger, enableHttp2))
+            .onError(err => Stream.eval(logger.warn(err)("Failed to upgrade socket to TLS")))
+            .mask
```

Comment:
Maybe we can even simplify it to this:
```suggestion
            .handleErrorWith(err => Stream.exec(logger.warn(err)("Failed to upgrade socket to TLS")))
```

Reply:
I thought so too but that does not compile because that log statement creates a `Stream[F, Unit]` which the compiler does not like

Reply:
ahhh no, I did not see the exec, thanks!

Reply:
Are you sure? That sounds like `Stream.eval`, not `Stream.exec`.

Reply:
I saw that the second after sending the comment, the way my brain works lots of the time :joy_cat: